### PR TITLE
feat(jans-cedarling): Add support of parsing the `Set` type in cedar-policy schema

### LINKED ISSUE #9836: feat(jans-cedarling): Add support of parsing the `Set` type in cedar-policy schema
**Is your feature request related to a problem? Please describe.**
In the current implementation of Cedarling, we cannot use the `Set` type in attributes within the Cedar policy schema. As a result, arrays cannot be used in the JWT token payload attributes.

**Describe the solution you'd like**
add parsing in the json cedar-policy schema `Set` type and implement building it from JWT attributes data.

**Additional context**
-

### PR DESCRIPTION
**Pull request by olehbozhok**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description
  Add parsing `Set` type from cedar-policy schema and implement build entity based on it.

#### Target issue
  
  [link](https://github.com/JanssenProject/jans/issues/9836)
  
  
  
closes #9836

#### Implementation Details
 Added parsing `Set` type from cedar-policy schema JSON and implement build entity based on it.
 Also implemented support nested `Set` type for example `Set<Set<String>>`
 
 Simplified file `cedarling\src\authz\token_data.rs`

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

File: jans-cedarling/cedarling/src/common/cedar_schema/cedar_json.rs [hunk 1]
```diff
@@ -62,50 +71,149 @@ impl CedarSchemaRecord {
...
 
 impl CedarSchemaEntityAttribute {
     pub fn is_required(&self) -> bool {
-        match self {
-            Self::Typed(entity_type) => entity_type.required.unwrap_or(true),
-            Self::Primitive(primitive_type) => primitive_type.required.unwrap_or(true),
+        self.required
+    }
+
+    pub fn get_type(&self) -> Option<CedarType> {
```
Comment (abaghinyan) [inline jans-cedarling/cedarling/src/common/cedar_schema/cedar_json.rs]: Should this function return `Option` or `Result` ?
Is the type mandatory ?

File: jans-cedarling/cedarling/src/authz/token_data.rs [hunk 2]
```diff
@@ -18,27 +17,19 @@ pub(crate) struct TokenPayload {
...
+    pub fn get(&'a self, key: &'a str) -> Result<Payload<'a>, GetTokenClaimValue> {
+        self.payload
+            .get(key)
+            .map(|value| Payload {
+                key: key.to_string(),
+                value,
+            })
+            .ok_or(GetTokenClaimValue::KeyNotFound(key.to_string()))
     }
 }
```
Comment (abaghinyan) [inline jans-cedarling/cedarling/src/authz/token_data.rs]: You can omit the lifetime for the return value and simplify it like this 
```
impl TokenPayload {
    fn new(payload: HashMap<String, serde_json::Value>) -> Self {
        Self { payload }
    }

    pub fn get(&self, key: &str) -> Result<Payload, GetTokenClaimValue> {
        self.payload
            .get(key)
            .map(|value| Payload {
                key: key.to_string(),
                value,
            })
            .ok_or(GetTokenClaimValue::KeyNotFound(key.to_string()))
    }
}
```

**Review by abaghinyan — CHANGES_REQUESTED** [review]
Some suggestions

**Review by rmarinn — APPROVED** [review]

**Review by duttarnab — APPROVED** [review]

File: jans-cedarling/cedarling/src/authz/token_data.rs (hunk 2 above)
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/authz/token_data.rs]: Thanks, resolved

File: jans-cedarling/cedarling/src/common/cedar_schema/cedar_json.rs (hunk 1 above)
Comment (olehbozhok) [reply jans-cedarling/cedarling/src/common/cedar_schema/cedar_json.rs]: You, are right. It is better to return error. 
Resolved

**Review by abaghinyan — APPROVED** [review]

**Review by duttarnab — APPROVED** [review]

**Pull request merged**